common/predictive-text/worker/model-compositor.ts

```diff
@@ -295,9 +272,38 @@ class ModelCompositor {
       suggestions = [ keepOption ].concat(suggestions);
     }
 
+    // Apply 'after word' punctuation.  We delay until now so that utility functions relying on the
+    // unmodified Transform may execute properly.
+    suggestions.forEach(function(suggestion) {
+      if (suggestion.transform.insert.length > 0) {
+        suggestion.transform.insert += punctuation.insertAfterWord;
+      }
+    });
+
     return suggestions;
   }
 
+  private toAnnotatedKeepSuggestion(suggestion: Suggestion & {p?: number}, noQuotes: boolean = false): Suggestion & {p?: number} {
+    let { open, close } = this.punctuation.quotesForKeepSuggestion;
+    
+    if(noQuotes) {
+      open = '';
+      close = '';
+    } else if(this.punctuation.isRTL) {
+      let temp = close;
+      close = open;
+      open = temp;
```

Comment:
I referred to: https://github.com/keymanapp/keyman/blob/1937ac80b42800da65502ec264b432ee6c28faa4/common/predictive-text/worker/model-compositor.ts#L253-L265

Reply:
https://github.com/keymanapp/keyman/blame/1937ac80b42800da65502ec264b432ee6c28faa4/common/predictive-text/worker/model-compositor.ts#L253-L265

![Spiderman pointing at spiderman](https://user-images.githubusercontent.com/2294397/95097802-20a99d00-06eb-11eb-9b91-77ab78e9aa31.jpg)

:joy: okay, but what is the _correct_ behaviour? When I initially created the terms "open" and "close" quote, I _intentionally_ avoided the "left" and "right" terminology, partially because I don't understand how this works in a right-to-left script, and because I was copying the [CSS specification](https://www.w3.org/TR/CSS2/generate.html#quotes). @mcdurdin, are you more familiar with RTL scripts to know how they're supposed to work?

Reply:
At this point, I'd also like to point out that this is technically a bit beyond the scope of this PR.  _This_ PR aims to refactor the _existing_ code patterns into something more maintainable moving forward.

As you've clearly noted here, we do need a proper "pass" over our RTL codepaths... but I don't believe that should be as a 'rider' on this PR.  It can easily be separated out into its own issue.  (As a side effect, the requested unit test is probably best implemented when "its own issue" is addressed.)

Reply:
I've refactored this site as noted in another comment, but left the "open" / "close" stuff the same for now.